common/remote/sockfile.cpp

```diff
@@ -3996,8 +4010,19 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer
         msg.read(name->text).read(mode).read(share);  
         // also try to recv extra byte
         byte extra = 0;
+        unsigned short sMode = IFUnone;
+        unsigned short cFlags = IFUnone;
         if (msg.remaining() >= sizeof(byte))
+        {
             msg.read(extra);
+            // and then try to recv extra sMode, cFlags
+            if (msg.remaining() >= sizeof(sMode))
+            {
+                msg.read(sMode);
+                if (msg.remaining() >= sizeof(cFlags))
```

Comment:
No point in this check is there? No client could send just sMode without also sending cFlags.